clippy_lints/src/utils/conf.rs

```diff
@@ -146,23 +146,8 @@ define_Conf! {
     (blacklisted_names, "blacklisted_names", ["foo", "bar", "baz", "quux"] => Vec<String>),
     /// Lint: CYCLOMATIC_COMPLEXITY. The maximum cyclomatic complexity a function can have
     (cyclomatic_complexity_threshold, "cyclomatic_complexity_threshold", 25 => u64),
-    /// Lint: DOC_MARKDOWN. The list of words this lint should not consider as identifiers needing ticks
-    (doc_valid_idents, "doc_valid_idents", [
-        "KiB", "MiB", "GiB", "TiB", "PiB", "EiB",
-        "DirectX",
-        "ECMAScript",
-        "GPLv2", "GPLv3",
-        "GitHub",
-        "IPv4", "IPv6",
-        "JavaScript",
-        "NaN",
-        "OAuth",
-        "OpenGL", "OpenSSH", "OpenSSL", "OpenStreetMap",
-        "TrueType",
-        "iOS", "macOS",
-        "TeX", "LaTeX", "BibTeX", "BibLaTeX",
-        "MinGW",
-    ] => Vec<String>),
+    /// Lint: DOC_MARKDOWN. The list of words this lint should not consider as identifiers needing ticks; words that are explicitly invalid can be prefixed with !
```

Comment:
I think this should talk about the default list and that `!` removes it from the list.

Also we could keep the default list here, and fill it into the lazy static at startup. Or is there any reason not to do that?

Reply:
Will do!

I think the issue is that this changes the meaning of this field; instead of being the list in its entirety, it's the difference from the default list. So, I don't think putting the list here makes sense.

Reply:
I think that's fine. The list shown in the docs will be the default list. Otherwise everybody will need to read code to figure out what things to disable.

I'm also wondering whether it's possible to generalize this disabling to the other lists

Reply:
Looking back at this, I think I'm going to modify this in a bit so that this logic is encapsulated in the `define_Conf!` macro.